### DISCUSSION TITLE
"Unable to read data from the transport connection: An existing connection was forcibly closed by the remote host."

### DISCUSSION THREAD

**Discussion by teSill** [jchristn/WatsonTcp#247] [Q&A]
WatsonTcp consistently catches this exception after updating to the latest update of the library. It happens whenever a standalone client is closed from the X button rather than gracefully logging out/disconnecting from within the game client.

The "ClientDisconnected" event is still called as expected and I'm able to process the disconnecting client's data as needed, so I'm not sure if this is really a problem. Is it safe to ignore or should I be taking some precautions against it?

**Comment by jchristn** [comment]
Yes, as of the latest release all of the exceptions propagate. There was an issue filed that showed where I was catching a series of different exceptions, indicated I disconnect event that we're just being silently thrown out.
